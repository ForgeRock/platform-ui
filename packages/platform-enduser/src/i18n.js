/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
