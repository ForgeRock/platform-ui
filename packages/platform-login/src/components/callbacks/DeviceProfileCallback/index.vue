<!--
Copyright (c) 2020 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details.
-->

<template>
  <div
    class="row justify-content-center">
    <FrSpinner
      label="Spinning"
      class="mb-4" />
    <div class="text-center text-muted w-100">
      {{ message }}
    </div>
  </div>
</template>

<script>
import Spinner from '@forgerock/platform-shared/src/components/Spinner/';
import { FRDevice } from '@forgerock/javascript-sdk';

export default {
  components: {
    FrSpinner: Spinner,
  },
  props: {
    callback: {
      type: Object,
      required: true,
    },
    nextStep: {
      type: Function,
      required: true,
    },
  },
  mounted() {
    this.message = this.callback.getMessage();
    this.deviceProfileCallback();
  },
  data() {
    return {
      message: '',
    };
  },
  methods: {
    deviceProfileCallback() {
      const device = new FRDevice();
      device.getProfile({
        location: this.callback.isLocationRequired(),
        metadata: this.callback.isMetadataRequired(),
      }).then((profile) => {
        this.callback.setProfile(profile);
        this.nextStep();
      }).catch(() => {
        this.callback.setProfile({});
        this.nextStep();
      });
    },
  },
};
</script>
