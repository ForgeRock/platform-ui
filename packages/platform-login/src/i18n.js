/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
