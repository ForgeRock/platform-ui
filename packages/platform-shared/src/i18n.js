/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
