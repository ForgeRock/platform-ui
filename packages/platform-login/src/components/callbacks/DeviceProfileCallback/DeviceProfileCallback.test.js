/* eslint-disable no-shadow */
/**
 * @license
 * Copyright (c) 2020 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import DeviceProfileCallback from '@/components/callbacks/DeviceProfileCallback';
import i18n from '@/i18n';

jest.mock('@forgerock/javascript-sdk', () => ({
  // eslint-disable-next-line object-shorthand
  FRDevice: function FRDevice() {
    return {
      getProfile() {
        return Promise.resolve({ identifier: 'hi' });
      },
    };
  },
}));

describe('DeviceProfileCallback', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(DeviceProfileCallback, {
      i18n,
      propsData: {
        callback: {
          getMessage: () => 'Message',
          isLocationRequired: () => true,
          isMetadataRequired: () => true,
          setProfile: jest.fn(),
        },
      },
    });
  });

  it('Loads DeviceProfileCallback component', () => {
    expect(wrapper.name()).toEqual('DeviceProfileCallback');
  });

  it('Sets profile data and emits "next-step"', async () => {
    expect(wrapper.vm.$data.message).toBe('Message');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$props.callback.setProfile).toHaveBeenCalledWith(expect.objectContaining({
      identifier: expect.any(String),
    }));
    expect(wrapper.emitted()['next-step'].pop()).toBeTruthy();
  });
});
