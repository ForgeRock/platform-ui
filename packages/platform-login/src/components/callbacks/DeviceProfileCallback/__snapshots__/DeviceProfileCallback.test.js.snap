// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceProfileCallback Loads DeviceProfileCallback component: console-error-snapshot 1`] = `
Array [
  Array [
    "Error: Not implemented: HTMLCanvasElement.prototype.getContext (without installing the canvas npm package)
    at module.exports (/node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)
    at HTMLCanvasElementImpl.getContext (/node_modules/jsdom/lib/jsdom/living/nodes/HTMLCanvasElement-impl.js:42:5)
    at HTMLCanvasElement.getContext (/node_modules/jsdom/lib/jsdom/living/generated/HTMLCanvasElement.js:130:58)
    at FRDevice.getInstalledFonts (/node_modules/@forgerock/javascript-sdk/src/fr-device/index.ts:145:49)
    at FRDevice.<anonymous> (/node_modules/@forgerock/javascript-sdk/src/fr-device/index.ts:221:23)
    at step (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:57:23)
    at Object.next (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:38:53)
    at /node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:32:71
    at new Promise (<anonymous>)
    at __awaiter (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:28:12)",
    undefined,
  ],
]
`;

exports[`DeviceProfileCallback Loads DeviceProfileCallback component: console-warn-snapshot 1`] = `
Array [
  Array [
    "Cannot generate profile ID. Crypto and/or getRandomValues is not supported.",
  ],
  Array [
    "Cannot collect font data. Browser does not support 2d canvas context",
  ],
  Array [
    "Cannot collect geolocation information. navigator.geolocation is not defined.",
  ],
]
`;

exports[`DeviceProfileCallback Sets profile data and emits "next-step": console-error-snapshot 1`] = `
Array [
  Array [
    "Error: Not implemented: HTMLCanvasElement.prototype.getContext (without installing the canvas npm package)
    at module.exports (/node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)
    at HTMLCanvasElementImpl.getContext (/node_modules/jsdom/lib/jsdom/living/nodes/HTMLCanvasElement-impl.js:42:5)
    at HTMLCanvasElement.getContext (/node_modules/jsdom/lib/jsdom/living/generated/HTMLCanvasElement.js:130:58)
    at FRDevice.getInstalledFonts (/node_modules/@forgerock/javascript-sdk/src/fr-device/index.ts:145:49)
    at FRDevice.<anonymous> (/node_modules/@forgerock/javascript-sdk/src/fr-device/index.ts:221:23)
    at step (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:57:23)
    at Object.next (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:38:53)
    at /node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:32:71
    at new Promise (<anonymous>)
    at __awaiter (/node_modules/@forgerock/javascript-sdk/lib/fr-device/index.js:28:12)",
    undefined,
  ],
]
`;

exports[`DeviceProfileCallback Sets profile data and emits "next-step": console-warn-snapshot 1`] = `
Array [
  Array [
    "Cannot generate profile ID. Crypto and/or getRandomValues is not supported.",
  ],
  Array [
    "Cannot collect font data. Browser does not support 2d canvas context",
  ],
  Array [
    "Cannot collect geolocation information. navigator.geolocation is not defined.",
  ],
]
`;
